vumi/middleware/base.py

```diff
@@ -176,7 +175,14 @@ class MiddlewareStack(object):
     """
 
     def __init__(self, middlewares):
-        self.middlewares = middlewares
+        self.consume_middlewares = self._sort_by_priority(
+            middlewares, 'consume_priority')
+        self.publish_middlewares = self._sort_by_priority(
+            reversed(middlewares), 'publish_priority')
+
+    @staticmethod
+    def _sort_by_priority(middlewares, priority_key):
+        return sorted(middlewares, key=lambda mw: mw.config[priority_key])
```

Comment:
Can you add a comment here about relying on Python's sorting algorithm being stable? It's not something that's likely to change, but it will help us understand how the ordering works when we look at this code again later.